[v2] mime sniffing local resources can result in browser evaluating them in wrong context

<!---
*** Please note: No new feature requests are being accepted for Wails v1 ***
--->

**Is your feature request related to a problem? Please describe.**
Just was confronted with the following scenario: clone pdf.js repository and build a local dist to serve the repo locally in wails following the instructions at https://github.com/mozilla/pdf.js to embed a local pdf viewer. The problem: 
- some HTML files like viewer.html have a leading BOM (utf-8 byte order mark) leading them to be characterized as `text/plain` instead of `text/html` with the result that the browser may display them as plain text
- some SVG files start with a comment such as `<!--- comment -->` leading them to be characterized as `text/html` instead of `image/svg+xml` with the result that the svg files may not be displayed properly.
  

**Describe the solution you'd like**
For html files we could check if BOM is present at the start of file and if so pass the slice after the BOM to the mime detect functions. For SVG files we could check for a comment section at the start of file and if present pass the slice after the end of the comment section. 

Alternatively, we could short-circuit detection of .html and .svg files in the same way that `mimecache.go` currently does in `mimecache.GetMimeType` for .js and .css files. 

For security and future compatibility reasons I'd prefer testing for the BOM in .html and for comments in .svg files and  allow `mimetype.detect` and `http.DetectContentType` to sniff the the mime-type.  

**Describe alternatives you've considered**
Manually remove BOM and eventual comments from resources is possible but I can think of situations where one could register additional (user-defined) custom uri schemes which would access resources other than embedded ones (for example from the file system at large) for display in the browser.

**Additional context**
Example where viewer.html starts with a BOM is embedded in an iframe :

![image](https://user-images.githubusercontent.com/6690378/160855944-02d9ce19-b54b-43a5-9676-f39e3366ad17.png)

Example where viewer.html BOM is removed/short-circuited but some SVG files with leading comments are passed as `text/html` instead of `image/svg+xml`:

![image](https://user-images.githubusercontent.com/6690378/160856652-eadb6aae-4e60-4b17-b5dc-b9dd718e91d1.png)

Example of what the viewer looks like with all .html and .svg files evaluated in the correct context:

![image](https://user-images.githubusercontent.com/6690378/160857254-4b956572-1473-4253-8a28-9719edcd0230.png)

Here's the result of the tests. With the current `github.com/gabriel-vasile/mimetype v1.3.1`:
```
--- FAIL: TestGetMimetype (0.00s)
    --- FAIL: TestGetMimetype/html-bom-utf8 (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/plain; charset=utf-8', want 'text/html; charset=utf-8'
    --- FAIL: TestGetMimetype/svg-w-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
    --- FAIL: TestGetMimetype/svg-w-control-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
FAIL
```
 with the latest current release v1.4.0:

```
--- FAIL: TestGetMimetype (0.00s)
    --- FAIL: TestGetMimetype/html-bom-utf8 (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/plain; charset=utf-8', want 'text/html; charset=utf-8'
    --- FAIL: TestGetMimetype/svg-w-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
    --- FAIL: TestGetMimetype/svg-w-control-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
FAIL
```

interestingly both versions pass the `svgWithBomCommentAndControlChars` test

the latest upstream (not yet released) only fails on html BOM:

```
--- FAIL: TestGetMimetype (0.00s)
    --- FAIL: TestGetMimetype/html-bom-utf8 (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/plain; charset=utf-8', want 'text/html; charset=utf-8'
FAIL
```
Proposal: @stffabi  and @leaanthony  - how about we merge the current PR and check/revert when the latest upstream is released?

Here's the result of the tests. With the current `github.com/gabriel-vasile/mimetype v1.3.1`:
```
--- FAIL: TestGetMimetype (0.00s)
    --- FAIL: TestGetMimetype/html-bom-utf8 (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/plain; charset=utf-8', want 'text/html; charset=utf-8'
    --- FAIL: TestGetMimetype/svg-w-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
    --- FAIL: TestGetMimetype/svg-w-control-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
FAIL
```
 with the latest current release v1.4.0:

```
--- FAIL: TestGetMimetype (0.00s)
    --- FAIL: TestGetMimetype/html-bom-utf8 (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/plain; charset=utf-8', want 'text/html; charset=utf-8'
    --- FAIL: TestGetMimetype/svg-w-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
    --- FAIL: TestGetMimetype/svg-w-control-comment (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/html; charset=utf-8', want 'image/svg+xml'
FAIL
```

interestingly both versions pass the `svgWithBomCommentAndControlChars` test

the latest upstream (not yet released) only fails on html BOM:

```
--- FAIL: TestGetMimetype (0.00s)
    --- FAIL: TestGetMimetype/html-bom-utf8 (0.00s)
        mimecache_test.go:42: GetMimetype() = 'text/plain; charset=utf-8', want 'text/html; charset=utf-8'
FAIL
```
Proposal: @stffabi  and @leaanthony  - how about we merge the current PR and check/revert when the latest upstream is released?

--- a/v2/go.mod
+++ b/v2/go.mod
@@ -6,7 +6,6 @@ require (
 	github.com/Masterminds/semver v1.5.0
 	github.com/flytam/filenamify v1.0.0
 	github.com/fsnotify/fsnotify v1.4.9
-	github.com/gabriel-vasile/mimetype v1.3.1
 	github.com/go-git/go-billy/v5 v5.2.0 // indirect
 	github.com/go-git/go-git/v5 v5.3.0
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
@@ -30,6 +29,7 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/tc-hib/winres v0.1.5
 	github.com/tidwall/sjson v1.1.7
+	github.com/wailsapp/mimetype v1.4.1-beta.1.0.20220331112158-6df7e41671fe
 	github.com/wzshiming/ctc v1.2.3
 	github.com/ztrue/tracerr v0.3.0
 	golang.org/x/mod v0.4.1

